pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -172,12 +171,10 @@ func getBlockVolumeIDToNodeUUIDMap(ctx context.Context, c *controller,
 
 	var vmRefs []types.ManagedObjectReference
 	var vmMoList []mo.VirtualMachine
-	properties := []string{"runtime.host", "config.hardware"}
+	properties := []string{"runtime.host", "config.hardware", "config.uuid"}
 
 	for _, nodeVM := range allnodeVMs {
-		vmRef := nodeVM.Reference().Value
 		vmRefs = append(vmRefs, nodeVM.Reference())
-		vmRefToUUID[vmRef] = nodeVM.UUID
 	}
 
 	for _, vc := range vCenters {
```

Comment:
In Line 186, the vmMoList gets overwritten and not appended to, when the 2nd, 3rd etc VC properties are retrieved right?

Reply:
`pc.Retrieve(ctx, vmRefs, properties, &vmMoList)` does not overwrite content of vmMoList